libia2/ia2.c

```diff
@@ -71,28 +71,37 @@ int protect_pages(struct dl_phdr_info *info, size_t size, void *data) {
   if (!lib)
     exit(-1);
 
-  struct AddressRange shared_ranges[NUM_SHARED_RANGES];
+  struct AddressRange shared_ranges[NUM_SHARED_RANGES] = {0};
+  struct AddressRange *cur_range = &shared_ranges[0];
   for (size_t i = 0; i < NUM_SHARED_RANGES; i++) {
     // Clear any potential old error conditions
     dlerror();
-    shared_ranges[i].start = (uint64_t)dlsym(lib, shared_sections[i][0]);
-    uint64_t aligned_start = shared_ranges[i].start & ~0xFFFUL;
-    if (aligned_start != shared_ranges[i].start) {
+
+    cur_range->start = (uint64_t)dlsym(lib, shared_sections[i][0]);
+    if (!cur_range->start) {
+      // We didn't find that symbol, oh well.
```

Comment:
I think it would be better to note that we didn't find the symbol because the section didn't exist--sections not existing is fine, but if the section did exist, not being able to find its start symbol would be a real problem.

Reply:
done